libmcount/dynamic.c

```diff
@@ -442,6 +443,9 @@ static void patch_func_matched(struct mcount_dynamic_info *mdi,
 
 	if (!found)
 		stats.nomatch++;
+
+	if (soname != NULL)
+		free(soname);
```

Comment:
It's ok to call free with a NULL argument.

By the way, this change itself has no meaning and you should move it to the actual use case.